shared/templates/file_groupowner/oval.template

```diff
@@ -13,23 +13,12 @@
    {{% endif %}}
     </criteria>
   </definition>
-  {{%- if MISSING_FILE_PASS -%}}
```

Comment:
This OVAL template now doesn't contain this parameter at all. How does it work now?

The docs says:

```
 -   **missing_file_pass** - If set to `"true"` the OVAL check will 
     pass when file is absent. Default value is `"false"`.
```

I have an impression that it now doesn't work when this is set to `false`. When it's set to `false` and the given file doesn't exist the rule should fail. But with this PR change if the given file doesn't exist, the `check_existence="none_exists"` is satisfied, therefore the rule will pass. Is my understanding correct?

Now the question is how to solve this change. I assume that the `missing_file_pass` set to `false` makes sense only when looking for specific files and not for recursive search. For the recursive search (i.e. rules with `recursive: true`) we usually don't care if something exists. But, our problem is mainly for recursively searching rules because in these rules we have the problem of collecting large amounts of item. We don't need this optimization for rules that match a single file or a couple of file in the same directory. Therefore we could have 2 versions of the OVAL code: one for recursive searching with the filter optimization and one for the other cases. We will document that the `missing_file_pass = false` is mutually exclusive with `recursive = true`. WDYT?

Reply:
We should remove this parameter. I didn't propose to remove from documentation because it should also be removed from many `rule.yml` files and I don't think this is mandatory for now since a behavior change is not expected. We can gradually deprecate this parameter for these affected templates.

These templates intend to assess the permissions and ownership (uid and gid) of files. Doesn't make any sense to me this assessment for files which are not present. If a file actually doesn't exist, it is quite logical that it won't have problems with permissions and ownership. The idea of the parameter `missing_file_pass` only makes sense for templates and rules which actually want to asses file existence and not their properties, like `file_existence` template or the `banner_etc_motd` rule, for example.

Also, seems that this parameter was introduced in these templates due to the design of the old OVAL logic where an empty file_object would affect the results based on the `check_existence=` parameter. In other words, this parameter is useless for these templates if we used `check_existence=none_exist` for the `file_test`.

Reply:
I agree with Jan here, the rule passing when a specific file is absent is a behaviour changes we do not want.
We have a specific parameter that allows this behaviour, and I think we should keep it, (if we want to remove it, it should be done in master). 

> Doesn't make any sense to me this assessment for files which are not present.

I think it depends on the expectation of the rule. There may be rules that don't care if it is missing, while others may care if the file is missing.
From my PoV, this is a flexibility of the template that we want to keep.